src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2563,7 +2564,15 @@ def update_site_appsettings(cmd, stack, site_config):
         NameValuePair = cmd.get_models('NameValuePair')
         if site_config.app_settings is None:
             site_config.app_settings = []
-        site_config.app_settings += [NameValuePair(name=k, value=v) for k, v in stack['configs'].items()]
+
+        for k, v in stack['configs'].items():
```

Comment:
Fix bug #15701 introduced by https://github.com/Azure/azure-cli/commit/3cd3b456dc5f18bca38700431cf3b8ca477d820d#diff-76cc90184fe3a5df817ac304d5929231658bef52748a4560dabf51ee18bf8f02R166. `match['setter']` adds to site_config.app_settings, but if the app setting already exists due to calling create on an existing app, it gets duplicated